pkg/controller/connectionsecret/connectionsecrets.go

```diff
@@ -196,13 +196,12 @@ func FillPrivateConnStrings(connStrings *mongodbatlas.ConnectionStrings, data *C
 }
 
 func GetAllServerless(ctx *workflow.Context, projectID string) ([]*mongodbatlas.Cluster, error) {
+	if IsCloudGovDomain(ctx) {
+		return make([]*mongodbatlas.Cluster, 0), nil
+	}
```

Comment:
is this a separate issue? I think this solves a separate issue, namely that serverless is not supported at all in cloud gov. Hence, I would like to see this in a separate contribution with a separate ticket.

Reply:
I wouldn't say it is a fix, more like an optimisation. There is no need to call "GetServerlessDeployments" if we're running on cloud-gov.